File: Models/Order.cs
using System;
using System.Collections.Generic;

namespace Backend.Models;

public partial class Order
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public decimal TotalPrice { get; set; }

    public int PaymentId { get; set; }

    public DateTime OrderDate { get; set; }

    public DateTime DeliveryDate { get; set; }

    public string? Note { get; set; }

    public string Status { get; set; } = null!;

    public virtual ICollection<OrderDetail> OrderDetails { get; set; } = new List<OrderDetail>();

    public virtual Payment Payment { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}

File: cms/src/types/order.ts
import { type BaseModel } from '@/base/model';

import { type Product } from './product';
import { type UserModel } from './user';

export interface Order extends BaseModel {
  userId: number;
  totalPrice: number;
  orderDate: string;
  deliveryDate: string;
  note: string;
  status: string;
  user: UserModel;
  orderDetails: OrderDetails[];
}

interface OrderDetails {
  orderId: number;
  productId: number;
  quantity: number;
  price: number;
  product: Product;
}
